src/main/java/org/opensearch/flowframework/template/TemplateParser.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.template;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.admin.indices.create.CreateIndexRequest;
+import org.opensearch.flowframework.workflow.WorkflowData;
+import org.opensearch.flowframework.workflow.WorkflowStep;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Utility class for parsing templates.
+ */
+public class TemplateParser {
+
+    private static final Logger logger = LogManager.getLogger(TemplateParser.class);
+
+    // Field names in the JSON. Package private for tests.
+    static final String WORKFLOW = "sequence";
+    static final String NODES = "nodes";
+    static final String NODE_ID = "id";
+    static final String EDGES = "edges";
+    static final String SOURCE = "source";
+    static final String DESTINATION = "dest";
+
+    /**
+     * Prevent instantiating this class.
+     */
+    private TemplateParser() {}
+
+    /**
+     * Parse a JSON representation of nodes and edges into a topologically sorted list of process nodes.
+     * @param json A string containing a JSON representation of nodes and edges
+     * @param workflowSteps A map linking JSON node names to Java objects implementing {@link WorkflowStep}
+     * @return A list of Process Nodes sorted topologically.  All predecessors of any node will occur prior to it in the list.
+     */
+    public static List<ProcessNode> parseJsonGraphToSequence(String json, Map<String, WorkflowStep> workflowSteps) {
+        Gson gson = new Gson();
```

Comment:
We can `ObjectMapper` from OpenSearch here instead of adding a new dependency for parsing json.

Reply:
`ObjectMapper` is `@opensearch.internal`.

Can we open a new issue to discuss this?  I don't want to delay this PR to have this discussion.

Reply:
We should be a little strict in introducing new dependencies in the plugin and fetch more from OpenSearch if already present. Yes, we should open an issue to discuss this.

Reply:
https://github.com/opensearch-project/opensearch-ai-flow-framework/issues/46